pkg/controllers/kube-controller-manager.go

```diff
@@ -17,88 +17,99 @@ package controllers
 
 import (
 	"context"
+	"encoding/json"
 	"errors"
+	"fmt"
+	"sort"
 
-	"github.com/spf13/cobra"
-
+	embedded "github.com/openshift/microshift/assets"
 	"github.com/openshift/microshift/pkg/assets"
 	"github.com/openshift/microshift/pkg/config"
 	"github.com/openshift/microshift/pkg/util"
 	"github.com/openshift/microshift/pkg/util/cryptomaterial"
 
-	"k8s.io/component-base/cli/globalflag"
-	"k8s.io/component-base/version/verflag"
+	kubecontrolplanev1 "github.com/openshift/api/kubecontrolplane/v1"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
 	kubecm "k8s.io/kubernetes/cmd/kube-controller-manager/app"
-	kubecmoptions "k8s.io/kubernetes/cmd/kube-controller-manager/app/options"
 
+	"github.com/ghodss/yaml"
 	klog "k8s.io/klog/v2"
 )
 
+const (
+	kcmDefaultConfigAsset = "components/kube-controller-manager/defaultconfig.yaml"
+)
+
 type KubeControllerManager struct {
-	kubecmOptions *kubecmoptions.KubeControllerManagerOptions
-	kubeconfig    string
-	kubeadmConfig string
+	args    []string
+	applyFn func() error
+
+	// TODO: report configuration errors immediately
+	configureErr error
 }
 
 func NewKubeControllerManager(cfg *config.MicroshiftConfig) *KubeControllerManager {
 	s := &KubeControllerManager{}
-	s.configure(cfg)
+	// TODO: manage and invoke the configure bits independently outside of this.
```

Comment:
It does feel like we have a gap in the API for the controller manager if configuration can return an error now. Let's work on that separately from these changes, though.